server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -137,6 +137,9 @@ public void close() {
                         chunkConsumer.streamClose();
                     }
                 });
+            } else {
+                // if content is needed it must be copied or retained in prepareRequest
+                request.close();
```

Comment:
Releasing content after `prepareRequest` but before `RestChannelConsumer.accept`

Reply:
This goes against the goal of having `prepareRequest` handle only headers, leaving any body handling until `RestChannelConsumer.accept`. Do we really need to drop this quite so early or can we do it at the end of this method?

Reply:
I do keep in mind that we want to remove content from prepareRequest. For the sake of PR size I would take one step at a time. Also a good opportunity for me to see how content is used in handlers.